models/demos/ttnn_resnet/tests/multi_device/test_ttnn_resnet50_performant.py

```diff
@@ -0,0 +1,444 @@
+    num_devices = 1 if isinstance(device, ttnn.Device) else device.get_num_devices()
+
+    padded_input_shape, input_mem_config, _ = ttnn.get_conv_padded_input_shape_and_mem_config(
+        device=device,
+        input_tensor=tt_inputs,
+        conv_config=tt_resnet50.conv1_config,
+        batch_size=tt_resnet50.batch_size,
+        height=tt_resnet50.conv1_output_height,
+        width=tt_resnet50.conv1_output_width,
+        in_channels=tt_resnet50.conv1_input_channels,
+        out_channels=tt_resnet50.conv1_output_channels,
+    )
+
+    inputs_padded = ttnn.to_torch(tt_inputs, device=device, mesh_composer=mesh_composer)
+    inputs_padded = inputs_padded.reshape(num_devices, 1, -1, inputs_padded.shape[-1])
+    inputs_padded = torch.nn.functional.pad(
+        inputs_padded,
+        (0, padded_input_shape[-1] - inputs_padded.shape[-1], 0, padded_input_shape[-2] - inputs_padded.shape[-2]),
+    )
+    tt_inputs_host = ttnn.from_torch(
+        inputs_padded, dtype=ttnn.bfloat16, layout=ttnn.ROW_MAJOR_LAYOUT, mesh_mapper=mesh_mapper
+    )
+    return tt_inputs_host, input_mem_config
+
+
+def setup_dram_sharded_input(device, tt_inputs, tt_resnet50, mesh_mapper, mesh_composer):
+    tt_inputs_host, input_mem_config = setup_l1_sharded_input(
+        device, tt_inputs, tt_resnet50, mesh_mapper, mesh_composer
+    )
+    dram_grid_size = device.dram_grid_size()
+    dram_shard_spec = ttnn.ShardSpec(
+        ttnn.CoreRangeSet(
+            {ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(dram_grid_size.x - 1, dram_grid_size.y - 1))}
+        ),
+        [
+            divup(tt_inputs_host.volume() // tt_inputs_host.shape[-1], dram_grid_size.x),
+            tt_inputs_host.shape[-1],
+        ],
+        ttnn.ShardOrientation.ROW_MAJOR,
+        False,
+    )
+    sharded_mem_config_DRAM = ttnn.MemoryConfig(
+        ttnn.TensorMemoryLayout.HEIGHT_SHARDED, ttnn.BufferType.DRAM, dram_shard_spec
+    )
+
+    return tt_inputs_host, sharded_mem_config_DRAM, input_mem_config
+
+
+@run_for_wormhole_b0()
+@pytest.mark.parametrize("device_params", [{"l1_small_size": 24576}], indirect=True)
+@pytest.mark.parametrize(
+    "device_batch_size, act_dtype, weight_dtype, math_fidelity",
+    ((16, ttnn.bfloat8_b, ttnn.bfloat8_b, ttnn.MathFidelity.LoFi),),
+)
+@pytest.mark.parametrize("enable_async", [True, False])
+def test_run_resnet50_inference(
+    device_mesh, use_program_cache, device_batch_size, act_dtype, weight_dtype, math_fidelity, enable_async
+):
+    if device_batch_size == 8:
+        pytest.skip("Skipping batch size 8 due to memory config issue")
+    if is_wormhole_b0() and device_batch_size == 20:
+        pytest.skip("Skipping batch size 20 for Wormhole B0 due to fitting issue")
+
+    for device in device_mesh.get_device_ids():
+        device_mesh.get_device(device).enable_async(enable_async)
+        device_mesh.get_device(device).enable_program_cache()
```

Comment:
Why is this even an api? When do we not want to enable program cache?

Reply:
This was copied from basically all other multi device models which enable program cache/async, though looking at the `use_program_cache` fixture it seems to already do this? We can review / clean up this code for all models in a separate pr if this is the case